From da1dbc11201017bff92f0079c1ce7db25b92a249 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Barnab=C3=A1s=20P=C5=91cze?= <pobrn@protonmail.com>
Date: Thu, 2 May 2024 03:24:35 +0200
Subject: [PATCH] treewide: fix C++20 compilation error wrt. designated
 initializers

C++20 introduced designated initializers similar to the ones found
in C99, however, in C++ designated initializers cannot be mixed
with non-designated initializers. GCC rejects mixed initializers
with an error.
---
 spa/include/spa/graph/graph.h               | 4 ++--
 spa/include/spa/pod/dynamic.h               | 2 +-
 spa/plugins/aec/aec-webrtc.cpp              | 2 +-
 spa/plugins/libcamera/libcamera-device.cpp  | 2 +-
 spa/plugins/libcamera/libcamera-manager.cpp | 2 +-
 spa/plugins/libcamera/libcamera-source.cpp  | 2 +-
 6 files changed, 7 insertions(+), 7 deletions(-)

diff --git a/spa/include/spa/graph/graph.h b/spa/include/spa/graph/graph.h
index 39049183a..117f6eef3 100644
--- a/spa/include/spa/graph/graph.h
+++ b/spa/include/spa/graph/graph.h
@@ -223,7 +223,7 @@ static inline int spa_graph_node_impl_sub_process(void *data SPA_UNUSED, struct
 }
 
 static const struct spa_graph_node_callbacks spa_graph_node_sub_impl_default = {
-	SPA_VERSION_GRAPH_NODE_CALLBACKS,
+	.version = SPA_VERSION_GRAPH_NODE_CALLBACKS,
 	.process = spa_graph_node_impl_sub_process,
 };
 
@@ -330,7 +330,7 @@ static inline int spa_graph_node_impl_reuse_buffer(void *data, struct spa_graph_
 }
 
 static const struct spa_graph_node_callbacks spa_graph_node_impl_default = {
-	SPA_VERSION_GRAPH_NODE_CALLBACKS,
+	.version = SPA_VERSION_GRAPH_NODE_CALLBACKS,
 	.process = spa_graph_node_impl_process,
 	.reuse_buffer = spa_graph_node_impl_reuse_buffer,
 };
diff --git a/spa/include/spa/pod/dynamic.h b/spa/include/spa/pod/dynamic.h
index 8cc9914e6..9edff2d05 100644
--- a/spa/include/spa/pod/dynamic.h
+++ b/spa/include/spa/pod/dynamic.h
@@ -41,7 +41,7 @@ static inline void spa_pod_dynamic_builder_init(struct spa_pod_dynamic_builder *
 		void *data, uint32_t size, uint32_t extend)
 {
 	static const struct spa_pod_builder_callbacks spa_pod_dynamic_builder_callbacks = {
-		SPA_VERSION_POD_BUILDER_CALLBACKS,
+		.version = SPA_VERSION_POD_BUILDER_CALLBACKS,
 		.overflow = spa_pod_dynamic_builder_overflow
 	};
 	builder->b = SPA_POD_BUILDER_INIT(data, size);
diff --git a/spa/plugins/aec/aec-webrtc.cpp b/spa/plugins/aec/aec-webrtc.cpp
index 8337d7ce8..bde9b578f 100644
--- a/spa/plugins/aec/aec-webrtc.cpp
+++ b/spa/plugins/aec/aec-webrtc.cpp
@@ -292,7 +292,7 @@ static int webrtc_run(void *object, const float *rec[], const float *play[], flo
 }
 
 static const struct spa_audio_aec_methods impl_aec = {
-	SPA_VERSION_AUDIO_AEC_METHODS,
+	.version = SPA_VERSION_AUDIO_AEC_METHODS,
 	.add_listener = NULL,
 	.init = webrtc_init,
 	.run = webrtc_run,
diff --git a/spa/plugins/libcamera/libcamera-device.cpp b/spa/plugins/libcamera/libcamera-device.cpp
index 948291135..56853a74a 100644
--- a/spa/plugins/libcamera/libcamera-device.cpp
+++ b/spa/plugins/libcamera/libcamera-device.cpp
@@ -239,7 +239,7 @@ static int impl_set_param(void *object,
 }
 
 static const struct spa_device_methods impl_device = {
-	SPA_VERSION_DEVICE_METHODS,
+	.version = SPA_VERSION_DEVICE_METHODS,
 	.add_listener = impl_add_listener,
 	.sync = impl_sync,
 	.enum_params = impl_enum_params,
diff --git a/spa/plugins/libcamera/libcamera-manager.cpp b/spa/plugins/libcamera/libcamera-manager.cpp
index c21281b33..c179e9ef3 100644
--- a/spa/plugins/libcamera/libcamera-manager.cpp
+++ b/spa/plugins/libcamera/libcamera-manager.cpp
@@ -352,7 +352,7 @@ impl_device_add_listener(void *object, struct spa_hook *listener,
 }
 
 static const struct spa_device_methods impl_device = {
-	SPA_VERSION_DEVICE_METHODS,
+	.version = SPA_VERSION_DEVICE_METHODS,
 	.add_listener = impl_device_add_listener,
 };
 
diff --git a/spa/plugins/libcamera/libcamera-source.cpp b/spa/plugins/libcamera/libcamera-source.cpp
index 3d9273419..8d91f76af 100644
--- a/spa/plugins/libcamera/libcamera-source.cpp
+++ b/spa/plugins/libcamera/libcamera-source.cpp
@@ -905,7 +905,7 @@ static int impl_node_process(void *object)
 }
 
 static const struct spa_node_methods impl_node = {
-	SPA_VERSION_NODE_METHODS,
+	.version = SPA_VERSION_NODE_METHODS,
 	.add_listener = impl_node_add_listener,
 	.set_callbacks = impl_node_set_callbacks,
 	.sync = impl_node_sync,
-- 
GitLab

